src/aof.c

```diff
@@ -1771,14 +1771,22 @@ void aofRemoveTempFile(pid_t childpid) {
 void aofUpdateCurrentSize(void) {
     struct redis_stat sb;
     mstime_t latency;
+    int aof_fd = server.aof_fd;
 
     latencyStartMonitor(latency);
-    if (redis_fstat(server.aof_fd,&sb) == -1) {
+    if (aof_fd == -1) {
+        /* Don't care if this fails: aof_fd will be -1 and redis_fstat handle that. */
+        aof_fd = open(server.aof_filename,O_RDONLY|O_NONBLOCK);
+    } 
+    if (redis_fstat(aof_fd,&sb) == -1) {
```

Comment:
any reason not to just change `fstat` into `stat`?
```suggestion
    if (redis_stat(server.aof_filename,&sb) == -1) {
```

Reply:
I tried to find their differences but found that they are almost the same. On the contrary, using `stat` can simplify the code, so I think your suggestion is acceptable, thank you.